src/Typeahead.svelte

```diff
@@ -51,12 +54,15 @@
 
   async function select() {
     const result = results[selectedIndex];
-
-    value = extract(result.original);
+    
+    const selectedValue = extract(result.original);
+    if (inputAfterSelect == 'clear') value = '';
+    if (inputAfterSelect == 'update') value = selectedValue;
 
     dispatch("select", {
       selectedIndex,
-      selected: value,
+      searched: value,
```

Comment:
Not sure if `searched` contains the correct value.

When I search "ca" and select "California", `searched` is "California" instead of "ca."

`value` has to be assigned to another variable before either clearing/updating.

```js
let searched = value;

if (inputAfterSelect == "clear") value = "";
if (inputAfterSelect == "update") value = selectedValue;

dispatch("select", {
  selectedIndex,
  searched,
  selected: selectedValue,
  original: result.original,
  originalIndex: result.index,
 });
```

Reply:
That was already done in the first push. (https://github.com/Amerlander/svelte-typeahead/blob/60476c3e46e04a09ccb2e8afbc13ab7f778c410e/src/Typeahead.svelte#L59)

So changing `selected` to `selectedValue` shauld have been already done in the first push too and I just missed this line then.

Reply:
At least in my project it works fine atm:
![image](https://user-images.githubusercontent.com/3764089/108606412-41cd0a80-73ba-11eb-8c88-b6d121faf02d.png)

Reply:
Ah, and you have `inputAfterSelect` set to `"keep"`?

Reply:
right

Reply:
ah, I got what you mean... yes, i change that.

Reply:
I saved `value` now to a new const and it works with `clear` as well as `keep` now.